src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -327,8 +334,8 @@ protected Object readResolve() throws ObjectStreamException {
                     conf.setScopesOverride(this.overrideScopes);
                     serverConfiguration = conf;
                 } else {
-                    OicServerManualConfiguration conf =
-                            new OicServerManualConfiguration(tokenServerUrl, authorizationServerUrl);
+                    OicServerManualConfiguration conf = new OicServerManualConfiguration(
+                            /* TODO */ "migrated", tokenServerUrl, authorizationServerUrl);
```

Comment:
already called out as a breaking change, users (with manual config) would first need to update to [4.355.v3a_fb_fca_b_96d4](https://github.com/jenkinsci/oic-auth-plugin/releases/tag/4.355.v3a_fb_fca_b_96d4) to pick up `SECURITY-3441` and then set the issuer before upgrading.

users with autoconfigure, will regardless be ok.

As we have called out the compatibility a lot of this code could be simplified to remove the data migration.  (So long as a user saves their configuration before upgrading, as the migration does not force a save)

Reply:
"migrated" because something is needed so the realm can at least be constructed, it will in all instances not be correct.
Without this the code would blow up, with Jenkins will start, but you won't be able to login unless you have disabled token validation, but you may have the "escapeHatch" so you can go in an fix it in the UI rather than messing with `config.xml` files on disk.

I thought this would be the lesser of two evils.  Probably could/should have used "needs_to_be_set_correctly" or something else to better convey this - but as it is called out, if people are blindly installing plugin updates flagged as breaking without reading release notes, I'm not overly concerned.